internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec_test.go

```diff
@@ -986,6 +986,125 @@ var _ = Describe("Validate clusterlogforwarderspec", func() {
+			gclOutput := loggingv1.OutputSpec{
+
+				Name: "gcl-out",
+				Type: "googleCloudLogging",
+				OutputTypeSpec: loggingv1.OutputTypeSpec{
+					GoogleCloudLogging: &loggingv1.GoogleCloudLogging{BillingAccountID: "billingAccountID"},
+				},
+			}
+
+			pruneHost := loggingv1.FilterSpec{
+				Name: "prune",
+				Type: loggingv1.FilterPrune,
+				FilterTypeSpec: loggingv1.FilterTypeSpec{
+					PruneFilterSpec: &pruneSpec,
+				},
+			}
+
+			forwarderSpec.Outputs = []loggingv1.OutputSpec{gclOutput}
+			forwarderSpec.Filters = []loggingv1.FilterSpec{pruneHost}
+
+			forwarderSpec.Pipelines = []loggingv1.PipelineSpec{
+				{
+					Name:       "gclPruneHost",
+					OutputRefs: []string{"gcl-out"},
+					InputRefs:  []string{loggingv1.InputNameApplication},
+					FilterRefs: []string{"prune"},
+				},
+			}
+			clfStatus.Outputs.Set("gcl-out", condReady)
+
+			verifyPipelines("custom-clf-name", &forwarderSpec, clfStatus)
+			Expect(clfStatus.Pipelines).To(HaveLen(1))
+			Expect(clfStatus.Pipelines).To(HaveKey("gclPruneHost"))
+			conds := clfStatus.Pipelines["gclPruneHost"]
+			Expect(conds).To(HaveCondition(loggingv1.ConditionReady, false, loggingv1.ReasonInvalid, "googleCloudLogging cannot prune `.hostname` field.+"))
+		},
+			Entry("when `in` prunes .hostname", loggingv1.PruneFilterSpec{In: []string{".hostname"}}),
+			Entry("when `notIn` prunes .hostname", loggingv1.PruneFilterSpec{NotIn: []string{".foo"}}),
+			Entry("when `in` prunes .hostname but `notIn` does not", loggingv1.PruneFilterSpec{In: []string{".hostname"}, NotIn: []string{".foo"}}),
+			Entry("when `notIn` prunes .hostname but `in` does not", loggingv1.PruneFilterSpec{In: []string{".foo"}, NotIn: []string{".foo"}}),
+			Entry("when `notIn` && `in` prunes .hostname", loggingv1.PruneFilterSpec{In: []string{".hostname"}, NotIn: []string{".foo"}}),
+		)
+
+		DescribeTable("valid prune and gcl pipeline", func(pruneSpec loggingv1.PruneFilterSpec) {
+			gclOutput := loggingv1.OutputSpec{
+
+				Name: "gcl-out",
+				Type: "googleCloudLogging",
+				OutputTypeSpec: loggingv1.OutputTypeSpec{
+					GoogleCloudLogging: &loggingv1.GoogleCloudLogging{BillingAccountID: "billingAccountID"},
+				},
+			}
+
+			pruneHost := loggingv1.FilterSpec{
+				Name: "prune",
+				Type: loggingv1.FilterPrune,
+				FilterTypeSpec: loggingv1.FilterTypeSpec{
+					PruneFilterSpec: &pruneSpec,
+				},
+			}
+
+			forwarderSpec.Outputs = []loggingv1.OutputSpec{gclOutput}
+			forwarderSpec.Filters = []loggingv1.FilterSpec{pruneHost}
+
+			forwarderSpec.Pipelines = []loggingv1.PipelineSpec{
+				{
+					Name:       "gclPruneHost",
+					OutputRefs: []string{"gcl-out"},
+					InputRefs:  []string{loggingv1.InputNameApplication},
+					FilterRefs: []string{"prune"},
+				},
+			}
+
+			clfStatus.Outputs.Set("gcl-out", condReady)
+
+			verifyPipelines("custom-clf-name", &forwarderSpec, clfStatus)
+			Expect(clfStatus.Pipelines["gclPruneHost"]).To(HaveCondition("Ready", true, "", ""))
```

Comment:
This is the contrary to the other comment.  I think this should be "ConditionFalse" because the condition is "there is an invalid prune filter" and it does not apply: ref https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ConditionStatus